File: csharp/SlimDownYourAggregates.Tests/Original/Events/BookPublishedEvent.cs
using SlimDownYourAggregates.Tests.Original.Core;
using SlimDownYourAggregates.Tests.Original.Entities;

namespace SlimDownYourAggregates.Tests.Original.Events;

public record BookPublishedEvent(BookId BookId, ISBN ISBN, Title Title, Author Author): IDomainEvent;

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/events/BookPublishedEvent.java
package io.eventdriven.slimdownaggregates.original.events;


import io.eventdriven.slimdownaggregates.original.core.IDomainEvent;
import io.eventdriven.slimdownaggregates.original.entities.Author;
import io.eventdriven.slimdownaggregates.original.entities.BookId;
import io.eventdriven.slimdownaggregates.original.entities.ISBN;
import io.eventdriven.slimdownaggregates.original.entities.Title;

public record BookPublishedEvent(
  BookId bookId,
  ISBN isbn,
  Title title,
  Author author
) implements IDomainEvent {
}
